core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/view/SemanticBrowserView.java

```diff
@@ -397,7 +399,20 @@ public void clean() {
     // No need to set focus.
     boolean restoreState = shouldSetFocus ? true : false;
     shouldSetFocus = false;
-    setInput(null);
+
+    // Ensure execution of setInput in UI thread.
+    if (Display.getCurrent() == null) {
+        EclipseUIUtil.displayAsyncExec(new Runnable() {
```

Comment:
I think is shouldn't be executed in async in order to have the same process as before but using the UI Thread

Reply:
@lredor any thought about this?

Reply:
Oups, I missed the Steve comment. So I use async because there is no real reason to change the input in sync mode. If this method is called not in the UI thread, the UI thread is probably always in progress. Async allows to to the input refresh "later".